lib/mongoose_push/service/fcm/pool_supervisor.ex

```diff
@@ -0,0 +1,49 @@
+defmodule MongoosePush.Service.FCM.Pool.Supervisor do
+  @moduledoc """
+  This module is a basic FCM pool supervisor that has a list of active workers - temporary solution before migrating to Sparrow with its own supervisors
+  """
+  use Supervisor, id: :fcm_pool_supervisor
+  require Logger
+  alias MongoosePush.Service.FCM.Pools
+
+  @spec start_link([Keyword.t()]) :: Supervisor.on_start()
+  def start_link(arg) do
+    Supervisor.start_link(__MODULE__, arg)
+  end
+
+  @impl true
+  def init(pools_configs) do
+    nested_list =
+      for {pool_name, pool_config} <- pools_configs do
+        Logger.info(~s"Starting FCM pool with API key #{filter_secret(pool_config[:key])}")
+        pool_size = pool_config[:pool_size]
+
+        pidgeon_workers =
+          Enum.map(1..pool_size, fn id ->
+            worker_name = Pools.worker_name(:fcm, pool_name, id)
+
+            Supervisor.Spec.worker(
+              Pigeon.GCMWorker,
+              [worker_name, pool_config],
+              id: worker_name
+            )
+          end)
+      end
+
+    list = List.flatten(nested_list)
```

Comment:
This name, and also the `nested_list` should be more descriptive. "list" doesn't tell anything about the variable other then type.